.github/workflows/binary-compatibility-checks.yml

```diff
@@ -0,0 +1,53 @@
+name: Binary Compatibility
+
+on:
+  pull_request:
+  push:
+    branches: [ main ]
+    tags-ignore: [ v* ]
+
+concurrency:
+  # Only run once for latest commit per ref and cancel other (previous) runs.
+  group: ci-${{ github.ref }}
+  cancel-in-progress: true
+
+permissions: { }
+
+jobs:
+  check-binary-compatibility:
+    name: Check / Binary Compatibility
+    runs-on: ubuntu-20.04
+    if: github.repository == 'apache/incubator-pekko-http'
+    strategy:
+      fail-fast: false
+      matrix:
+        SCALA_VERSION: [ 2.12, 2.13, 3.3 ]
+        JDK: [ 8, 11 ]
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v4
+        with:
+          fetch-depth: 0
+          fetch-tags: 0
+
+      - name: Setup Java 11
+        uses: actions/setup-java@v3
+        with:
+          distribution: temurin
+          java-version: 11
```

Comment:
if you really need to test multiple JDKS, you need to not hardcode '11' - should use `${{ matrix.JDK }}`